src/js/components/AppListComponent.jsx

```diff
@@ -314,22 +366,21 @@ var AppListComponent = React.createClass({
       appsHealthCount: getInitialFilterCounts(HealthStatus)
     };
 
-    // Global search view - only display filtered apps
+    // Global search view - only display filtered apps & groups
     if (this.hasFilters()) {
       appListViewType = AppListViewTypes.APP_LIST;
 
-      nodesSequence = this.filterNodes(lazy(state.apps), filterCounts)
+      nodesSequence = lazy(this.filterNodes(state.apps, filterCounts))
```

Comment:
Ok, now I get it... so indeed, I got it wrong ;)
>  There is a lazy.js-concat existent.

Yes, but the `.filter` function will be executed asynchronously, and therefore the `.concat( groupNodes )` statement after the function will just concatenate an empty array ;)

I was looking for a function that can somehow be called when the expression is to be evaluated. Since I didn't know about `lazy.js` before, I had a quick look on the documentation and I found the `.reduce` function. However, when I tried the following, it triggered a chain of errors later in the execution:

```js
nodesSequence = nodesSequence.reduce((memo, app) => {
    // Push filtered items & new groups in memo
    memo.push( app );
    return memo;
}, [] );
```

Therefore, due to lack of time I decided to wrap-up my changes using non-lazy evaluation. 
But I can give this a try later tonight and change my pull request.